lib/vquic/quiche.c

```diff
@@ -773,28 +870,44 @@ static CURLcode http_request(struct Curl_easy *data, const void *mem,
     }
   }
 
+  *written = (ssize_t) len;
+
   switch(data->state.httpreq) {
   case HTTPREQ_POST:
   case HTTPREQ_POST_FORM:
   case HTTPREQ_POST_MIME:
   case HTTPREQ_PUT:
-    if(data->state.infilesize != -1)
-      stream->upload_left = data->state.infilesize;
-    else
+    if(data->state.infilesize != -1) {
+      stream->upload_left = len - ((char *)hdbuf - (char *)mem);
```

Comment:
This changes how `stream->upload_left` is used to now be how much memory there is in the current buffer to send. Is that on purpose?

Looking at this closer, we don't even use `->upload_left` elsewhere in quiche.c. Should we just use a local variable there instead?